applications/StructuralMechanicsApplication/tests/cpp_tests/test_mpc_plate_tension.cpp

```diff
@@ -0,0 +1,178 @@
+    //Create Nodes
+    auto pnode1 = model_part.CreateNewNode(1, 0.00,0.00,0.00);
+    auto pnode2 = model_part.CreateNewNode(2, 1.00,0.00,0.00);
+    auto pnode3 = model_part.CreateNewNode(3, 1.00,1.00,0.00);
+    auto pnode4 = model_part.CreateNewNode(4, 0.00,1.00,0.00);
+    auto pnode5 = model_part.CreateNewNode(5, 0.50,0.00,0.00);
+    auto pnode6 = model_part.CreateNewNode(6, 1.00,0.50,0.00);
+    auto pnode7 = model_part.CreateNewNode(7, 0.50,1.00,0.00);
+    auto pnode8 = model_part.CreateNewNode(8, 0.50,0.50,0.00);
+    //Create Elements
+    model_part.CreateNewElement("SmallDisplacementElement2D4N", 1, {{1, 5, 7, 4}}, p_Properties);
+    model_part.CreateNewElement("SmallDisplacementElement2D4N", 2, {{5, 2, 6, 8}}, p_Properties);
+    model_part.CreateNewElement("SmallDisplacementElement2D4N", 3, {{8, 6, 3, 7}}, p_Properties);
+    //Add DOF
+    VariableUtils().AddDofWithReaction(DISPLACEMENT_X, REACTION_X, model_part);
+    VariableUtils().AddDofWithReaction(DISPLACEMENT_Y, REACTION_Y, model_part);
+    // Initialize elements
+    auto& r_process_info = model_part.GetProcessInfo();
+    for (auto& elem : model_part.Elements())
+    {
+        elem.Initialize();
+        elem.InitializeSolutionStep(r_process_info);
+    }
+    // Set initial solution
+    for (auto& node : model_part.Nodes())
+    {
+        (node.FastGetSolutionStepValue(DISPLACEMENT)).clear();
+    }
+    //Apply BC
+    pnode1->Fix(DISPLACEMENT_X);
+    pnode1->Fix(DISPLACEMENT_Y);
+    pnode2->Fix(DISPLACEMENT_Y);
+    pnode4->Fix(DISPLACEMENT_X);
+    //Creat Load
+    auto p_condition1 = model_part.CreateNewCondition("LineLoadCondition2D2N", 1, std::vector<IndexType>({2,6}), p_Properties);
+    auto p_condition2 = model_part.CreateNewCondition("LineLoadCondition2D2N", 2, std::vector<IndexType>({6,3}), p_Properties);
+    array_1d<double, 3> load;
+    load[0] = 1000.0; load[1] = 0.0; load[2] = 0.0;
+    p_condition1->SetValue(LINE_LOAD, load);
+    p_condition2->SetValue(LINE_LOAD, load);
+    //Apply Constraints
+    std::vector< Dof<double>::Pointer >slave_dofs(2);
+    slave_dofs[0] = pnode8->pGetDof(DISPLACEMENT_X);
+    slave_dofs[1] = pnode8->pGetDof(DISPLACEMENT_Y);
+    std::vector< Dof<double>::Pointer >master_dofs(4);
+    master_dofs[0] = pnode7->pGetDof(DISPLACEMENT_X);
+    master_dofs[1] = pnode5->pGetDof(DISPLACEMENT_X);
+    master_dofs[2] = pnode7->pGetDof(DISPLACEMENT_Y);
+    master_dofs[3] = pnode5->pGetDof(DISPLACEMENT_Y);
+    Matrix relation_matrix = ZeroMatrix(2,4);
+    relation_matrix(0,0) = 0.5;
+    relation_matrix(0,1) = 0.5;
+    relation_matrix(1,2) = 0.5;
+    relation_matrix(1,3) = 0.5;
+    Vector constant_vector = ZeroVector(1);
+    model_part.CreateNewMasterSlaveConstraint("LinearMasterSlaveConstraint", 1, master_dofs, slave_dofs, relation_matrix, constant_vector);
+    //Setting up System
+    SchemeType::Pointer p_scheme = SchemeType::Pointer( new ResidualBasedIncrementalUpdateStaticSchemeType() );
```

Comment:
please use `Kratos::make_shared` instead of `new`

Reply:
what is the benefit of Kratos::make_shared? I guess to avoid a copy?

Reply:
yes in a nutshell :)
https://stackoverflow.com/questions/18301511/stdshared-ptr-initialization-make-sharedfoo-vs-shared-ptrtnew-foo

also because we have widely adapted this style now